src/new/sendTransaction.ts

```diff
@@ -422,6 +424,23 @@ export default class SendTransaction extends EventEmitter {
               throw new WalletError(ErrorMessages.TRANSACTION_IS_NULL);
             }
             this.transaction.updateHash();
+            if (this.storage) {
+              // Add transaction to storage and process storage
+              (async (storage: IStorage, transaction: Transaction) => {
+                // Get the transaction as a history object
+                const historyTx = await transactionUtils.convertTransactionToHistoryTx(
+                  transaction,
+                  storage
+                );
+                // Add transaction to storage
+                await storage.addTx(historyTx);
+                // Add token from a create token transaction to the storage
+                // This just returns if the transaction is not a CREATE_TOKEN_TX
+                await addCreatedTokenFromTx(transaction as CreateTokenTransaction, storage);
+                // Process new transaction
+                await storage.processNewTx(historyTx);
```

Comment:
Is it possible to call the `wallet.onNewTx` method instead of this? I think you are missing a call to `await this.scanAddressesToLoad();`. My main concern here is that we add things in that method and forget to add here as well, like this scan addresses to load.

Reply:
The issue is that the SendTransaction does not have the wallet instance.
We can change this to use the wallet instance instead of the storage instance, but it would require some breaking changes.

While the `scanAddressesToLoad` is missing from this logic it will run as soon as the tx is received via websocket.